### PR TITLE
Update output page display

### PR DESCRIPTION
**Pull request by MichaelPesce**
- update output page to display data in a table instead of accordions
- update input accordions to stack in 2 separate columns

addresses issue #132

### CODE REVIEW

**Comment by MichaelPesce** [comment]
new output display:
Screenshot 2024-09-05 at 11 34 51 AM

**Comment by MichaelPesce** [comment]
for comparison, this is what it looked liked prior: 
Screenshot 2024-09-05 at 4 33 03 PM

**Comment by MichaelPesce** [comment]
@avdudchenko 
Screenshot 2024-09-27 at 3 34 57 AM

File: electron/ui/src/App.js [hunk 1]
```diff
@@ -21,15 +23,21 @@ function App() {
     const [checkAgain, setCheckAgain] = useState(1)
     const WAIT_TIME = 2
 
-    console.log("App hasTheme = ",hasTheme);
+    const mui_theme = createTheme({
```
Comment (avdudchenko) [inline electron/ui/src/App.js:26]: can we add an note what this does

File: electron/ui/src/components/SingleOutput/SingleOutput.js [hunk 2]
```diff
@@ -1,30 +1,59 @@
...
     const {outputData, downloadOutput, saveConfig} = props;
     const [configName, setConfigName] = useState(outputData.name)
     const [openSaveConfig, setOpenSaveConfig] = React.useState(false);
     const [saved, setSaved] = React.useState(false);
+    const [outputTableData, setOutputTableData] = useState({})
 
     const handleOpenSaveConfig = () => setOpenSaveConfig(true);
     const handleCloseSaveConfig = () => setOpenSaveConfig(false);
 
+    useEffect(()=> {
```
Comment (avdudchenko) [inline electron/ui/src/components/SingleOutput/SingleOutput.js:34]: Also lets add a small note on what this does.

File: electron/ui/src/components/SingleOutput/SingleOutput.js [hunk 3]
```diff
@@ -87,101 +116,46 @@ export default function SingleOutput(props) {
...
-                            {value.display_name}
-                        </AccordionSummary>
-                        <AccordionDetails>
-                            <Box
-                                component="form"
-                                sx={{
-                                    '& > :not(style)': {m: 1},
-                                }}
-                                autoComplete="off"
+    const renderRows = () => {
```
Comment (avdudchenko) [inline electron/ui/src/components/SingleOutput/SingleOutput.js:119]: Can we also add a note on what we are doing here.

File: electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js [hunk 4]
```diff
@@ -234,41 +234,107 @@ export default function ConfigInput(props) {
...
-        //sorting the keys of var_sections by amount of variables into object "items"
-        let items = Object.keys(var_sections).map(function (key) {
-            return [key, var_sections[key]['num_variables']];
-        });
-        items.sort(function (first, second) {
-            return second[1] - first[1];
-        });
-        // console.log(items)
-        return var_sections
+        // sort the keys of var_sections into two groups that have as close as possible to even amount of total variables
```
Comment (avdudchenko) [inline electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js:237]: I assume this is used to calcualte spacing for the table?

**Review by avdudchenko — APPROVED** [review]
Just needs some comments to clarify what new code does. Otherwise, I like the new table look.

File: electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js [hunk 5]
```diff
@@ -234,41 +234,107 @@ export default function ConfigInput(props) {
...
+            }
+        } catch(e) {
+            console.log("error sorting: ")
+            console.log(e)
+        }
+        
+        return [var_sections_left, var_sections_right]
     }
 
     const renderInputAccordions = () => {
```
[the remark this answers is not in the document]
Comment (MichaelPesce) [reply electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js:285]: see [here](https://github.com/watertap-org/watertap-ui/blob/eb7313edde5c7b8f1bfed6b7a55317a095f7a14d/electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js#L179-L183)

File: electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js (hunk 4 above)
Comment (MichaelPesce) [reply electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js]: not exactly. it's to try to generate the input accordions into two columns that are close to even in height. in order to do so, we need to count how many input variables are in each category

File: electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js [hunk 6]
```diff
@@ -234,41 +234,107 @@ export default function ConfigInput(props) {
...
+        let var_sections_right = {}
+        let total_variables_left = 0
+        let total_variables_right = 0
+        let next_section = "left"
+        try {
+            for (let category of Object.keys(var_sections)) {
+                let section = var_sections[category]
+                let input_data = section.input_variables
+
+                // calculate variable amount - fixed variables take up about 45% as much space as free
```
[the remark this answers is not in the document]
Comment (MichaelPesce) [reply electron/ui/src/views/FlowsheetConfig/ConfigInput/ConfigInput.js:251]: this is referring to the amount of space that input variables take up on screen. fixed variables take up roughly half the amount of height as free variables. see below image:
Screenshot 2024-10-03 at 9 10 36 AM

I am attempting to make the columns of input accordions close to even in height when the input page loads

**Pull request merged**